llvm/lib/Analysis/ConstraintSystem.cpp

```diff
@@ -52,9 +52,6 @@ bool ConstraintSystem::eliminateUsingFM() {
   for (unsigned R1 = 0; R1 < NumRemainingConstraints; R1++) {
     // FIXME do not use copy
     for (unsigned R2 = R1 + 1; R2 < NumRemainingConstraints; R2++) {
-      if (R1 == R2)
-        continue;
```

Comment:
might be worth adding an assert initially?

Reply:
It is safe to remove this because `R1` and `R2` are both not modified inside the loop.

Reply:
Ah yes, thanks for the cleanup!